core/src/main/java/com/opensymphony/xwork2/config/providers/XmlConfigurationProvider.java

```diff
@@ -164,13 +174,10 @@ public boolean equals(Object o) {
         if (this == o) {
             return true;
         }
-
         if (!(o instanceof XmlConfigurationProvider)) {
             return false;
         }
-
-        final XmlConfigurationProvider xmlConfigurationProvider = (XmlConfigurationProvider) o;
-
+        XmlConfigurationProvider xmlConfigurationProvider = (XmlConfigurationProvider) o;
```

Comment:
```suggestion
        final XmlConfigurationProvider xmlConfigurationProvider = (XmlConfigurationProvider) o;
```

this could remain final

Reply:
I find it clutters the code and goes against [present common practice](https://stackoverflow.com/questions/154314/when-should-one-use-final-for-method-parameters-and-local-variables) - but I can revert these changes if that is the preferred code style.

Reply:
https://stackoverflow.com/questions/154314/when-should-one-use-final-for-method-parameters-and-local-variables/154510#154510 LGTM, but for this existing line "If I'm in someone else's code, I'm not going to pull them out but if I'm writing new code I won't put them in." would fit.

With your explanation, both styles would be ok for me. Maybe @lukaszlenart has some recommendation according to the Struts coding style?

Reply:
I'm fine with dropping `final` in this context, I like using `final`s for fields and function arguments, where it makes sense :)